.github/workflows/build.yaml

```diff
@@ -0,0 +1,49 @@
+name: Build and verify
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches: [master]
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        include:
+          - jdk: 11.0.16
+            sonarEnabled: true
+          - jdk: 17.0.4
+            sonarEnabled: false
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3.0.2
+        with:
+          # Fetch entire history for SonarCloud analysis.
+          fetch-depth: 0
+      - name: Set up JDK
+        uses: actions/setup-java@v3.4.1
+        with:
+          java-version: ${{ matrix.jdk }}
+          distribution: temurin
+          cache: maven
+      - name: Cache SonarCloud packages
+        if: ${{ matrix.sonarEnabled }}
+        uses: actions/cache@v3.0.8
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Display environment details and build the project
```

Comment:
It would be nice if we keep this in sync with the build from EPS. So I'd suggest splitting up showing the details and the actual build. Will push a proposal.